src/node/miner.cpp

```diff
@@ -383,7 +384,8 @@ void BlockAssembler::addPackageTxs(int& nPackagesSelected, int& nDescendantsUpda
     // and modifying them for their already included ancestors
     UpdatePackagesForAdded(inBlock, mapModifiedTx);
 
-    CTxMemPool::indexed_transaction_set::index<ancestor_score>::type::iterator mi = m_mempool.mapTx.get<ancestor_score>().begin();
+    // ELEMENTS: we use confidential_score instead of ancestor_score
+    CTxMemPool::indexed_transaction_set::index<confidential_score>::type::iterator mi = m_mempool.mapTx.get<confidential_score>().begin();
```

Comment:
This might be working as intended, but just to be clear, what we are doing here is *unconditionally* altering the block production function to sort by discounted weight, on every chain, even those chains with `acceptdiscount = false`!

Reply:
`CTxMemPoolEntry::GetDiscountTxSize()` fixed to be conditional, and added into the ordering test to check that nodes that don't accept discounted CT have their ordering as per ancestor package fee rate